### PR TITLE
CLOUDP-136890: Add auditing

### PR DESCRIPTION
**Pull request by fabritsius**
This PR adds `Auditing` to the project spec but also makes a couple changes to the `e2e` tests, including:
- add a generic function `WaitForConditionsToBecomeTrue` for checking conditions
- fix `integrations` test afterwards
- fix `X.509` test afterwards

### CODE REVIEW

**Comment by fabritsius** [comment]
Tests will have to be rebased on top of #683 so will push them later.

**Comment by fabritsius** [comment]
This PR finally passes all of the tests, so after a small tidy up it can be merged 👌

File: test/e2e/auditing_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	By("Remove Auditing from the project", func() {
+		userData.Project.Spec.Auditing = nil
+		Expect(userData.K8SClient.Update(userData.Context, userData.Project)).Should(Succeed())
+		actions.WaitForConditionsToBecomeTrue(userData, status.ReadyType)
+	})
+}
+
+func exampleFilter() string {
+	return "{\"atype\" : \"authenticate\", \"param\" : {\"user\" : \"auditReadOnly\", \"db\" : \"admin\", \"mechanism\" : \"SCRAM-SHA-1\"} }"
```
Comment (Sugar-pack) [inline test/e2e/auditing_test.go:76]: You can use this option to ignore special characters. https://go.dev/play/p/8IrA_Uv29cl
```
func exampleFilter() string {
	return `{"atype" : "authenticate", "param" : {"user" : "auditReadOnly", "db" : "admin", "mechanism" : "SCRAM-SHA-1"} }`
}
```

Comment (fabritsius) [reply test/e2e/auditing_test.go:76]: Sure, things like this will be in the upcoming tidy up commit :)

Comment (fabritsius) [reply test/e2e/auditing_test.go]: Done ✅

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍

**Pull request merged**